src/FilenameParsing/LegacyFileIDHelper.php

```diff
@@ -132,18 +132,34 @@ public function parseFileID($fileID)
             return null;
         }
 
+        // This helper is only used if the natural and hash helpers have already failed to parse the ID
+        // So if we can parse the ID here it means the project has a deprecated legacy file structure.
+        $deprecationMessage = 'Use of legacy file resolution strategies is deprecated.'
+        . ' See https://docs.silverstripe.org/en/4/developer_guides/files/file_migration/.'
+        . ' After migrating your files, change your file resolution configuration to match the defaults:'
+        . ' https://github.com/silverstripe/silverstripe-installer/blob/4/app/_config/assets.yml';
+
+
         // Can't have a resampled folder without a variant
         if (empty($matches['variant']) && strpos($fileID ?? '', '_resampled') !== false) {
-            return $this->parseSilverStripe30VariantFileID($fileID);
+            $parsed = $this->parseSilverStripe30VariantFileID($fileID);
+            if ($parsed) {
+                Deprecation::notice('4.13.0', $deprecationMessage, Deprecation::SCOPE_GLOBAL);
```

Comment:
Should we be adding what the fileID is if it `$parsed` to the Deprecation message?  $fileID it will be a part of URL right?  If these messages show up in their logs it would be helpful to try and track down what's actually linking to them?

Reply:
I'm not sure it would be helpful... the migration docs should take care of any files that are in a legacy setup without having to deal with each one individually, and we de-dup deprecation warnings based on the message so if a project has a fully legacy assets system with thousands of assets they'll end up with a FLOOD in their logs with one message for each file.

I'll add it in if you still think it'd be useful with that context in mind.